library/swift

```diff
@@ -2,7 +2,10 @@ Maintainers: Haris Amin <aminharis7@gmail.com> (@hamin),
              Thomas Catterall <me@swizzlr.co> (@swizzlr)
 GitRepo: https://github.com/swiftdocker/docker-swift.git
 
-Tags: 3.1.0, 3.1, 3, latest
+Tags: 4.0, 4, latest
+GitCommit: 84d2f69c035204bc2dc077f330deb7a393255a18
+
+Tags: 3.1.0, 3.1, 3
 GitCommit: ef9aa534705fc8ab4258c539f6304072ebae9613
 
 Tags: 3.0.2, 3.0
```

Comment:
Any particular reason to keep these older versions listed here?  Are they still actively supported upstream?  For example, will there ever be a 3.0.3 or 3.1.1 release?

Removing them here will remove them from the "Supported" section on the Hub readme (and will prevent us from spending cycles rebuilding it on the official build server), but the tag will still be available to pull for users who want/need it.  (See https://github.com/docker-library/official-images#library-definition-files for more detail on this.)

Reply:
@tianon technically they are supported and ppl still might be using them for various purposes before transitioning to 4.x . I think we'd like to keep them there if its not too much trouble

Reply:
"For example, will there ever be a 3.0.3 or 3.1.1 release?"

Not sure at the moment. Since that's a bit on apple's side. There might be security related patch releases that we don't know of and can't predict at the time being.

Reply:
Yeah, that's fair and fine -- probably would be a good idea to move those to either separate branches or separate directories so that they're easier to update if/when that time comes. :+1:

Reply:
@tianon awesome! And i've thought about reorganizing it a bit more via branches and/or directories. I like the latter approach, I've seen the redis image use that. Any additional reading/advice on that for future improvements would be greatly appreciated. Thanks again :)

Reply:
Having done both, I definitely recommend directories for something simple like this -- it's much easier to do things like `vimdiff` across versions that way, and you could then simplify your file to only reference a single `GitCommit` if you felt so inclined (just move the `GitCommit` field up to that top section with `GitRepo`, etc).

Also, speaking of, there already appears to be a 3.1.1 that's probably worth doing something with: https://swift.org/download/ :smirk:

Reply:
@tianon I added 3.1.1 here https://github.com/docker-library/official-images/pull/3535